bcda/models/models.go

```diff
@@ -395,10 +415,37 @@ func (cclfBeneficiary *CCLFBeneficiary) GetBlueButtonID(bb client.APIClient) (bl
 		return "", err
 	}
 
+	if len(patient.Entry) == 0 {
+		err = fmt.Errorf("patient identifier not found at BlueButton for cclfBenficiary ID: %v", cclfBeneficiary.ID)
+		log.Error(err)
+		return "", err
+	}
+	var foundHicn = false
+	var foundBlueButtonID = false
 	blueButtonID = patient.Entry[0].Resource.ID
-	fullURL := patient.Entry[0].FullUrl
-	if !strings.Contains(fullURL, blueButtonID) {
-		err = fmt.Errorf("patient identifier not found in the URL")
+	for _, identifier := range patient.Entry[0].Resource.Identifier {
+		if strings.Contains(identifier.System, "hicn-hash") {
+			if identifier.Value != hashed_hicn {
+				foundHicn = false
```

Comment:
probably could rewrite this to only catch the true case since foundHicn is already set to false. but not a deal breaker

Reply:
I felt that this was more readable.  Also, if it is somehow in there both ways this will likely fail which I think we want.

Reply:
Changed.